types/mytime.go

```diff
@@ -119,7 +119,25 @@ func (t MysqlTime) GoTime(loc *gotime.Location) (gotime.Time, error) {
 
 // IsLeapYear returns if it's leap year.
 func (t MysqlTime) IsLeapYear() bool {
-	return (t.year%4 == 0 && t.year%100 != 0) || t.year%400 == 0
+	return isLeapYear(t.year)
+}
+
+func isLeapYear(year uint16) bool {
+	return (year%4 == 0 && year%100 != 0) || year%400 == 0
+}
+
+var daysByMonth = [12]int{31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31}
+
+// GetLastDay returns the last day of the month
+func GetLastDay(year, month int) int {
+	var day = 0
+	if month > 0 {
```

Comment:
should we also check `month <= 12`?

Reply:
I've modified , and moved this PR to [PR8988](https://github.com/pingcap/tidb/pull/8988) (merge into master first ), as shenli suggested .
review at PR8988 ,and help me close this PR. thanks!